tests/activity_api_batch_unpause_test.go

```diff
@@ -0,0 +1,249 @@
+		description, err := s.SdkClient().DescribeWorkflowExecution(ctx, workflowRun1.GetID(), workflowRun1.GetRunID())
+		assert.NoError(t, err)
+		if description.GetPendingActivities() != nil {
+			assert.Len(t, description.PendingActivities, 1)
+			assert.True(t, description.PendingActivities[0].Paused)
+		}
+	}, 5*time.Second, 100*time.Millisecond)
+
+	// yes, both workflow type and activity type are "func1". There is no way to specify them.
+	workflowTypeName := "WorkflowFunc"
+	activityTypeName := "ActivityFunc"
+	// Make sure the activity is in visibility
+	var listResp *workflowservice.ListWorkflowExecutionsResponse
+	unpauseCause := fmt.Sprintf("%s = 'property:activityType=%s'", searchattribute.TemporalPauseInfo, activityTypeName)
+	query := fmt.Sprintf("(WorkflowType='%s' AND %s)", workflowTypeName, unpauseCause)
+
+	s.EventuallyWithT(func(t *assert.CollectT) {
+		listResp, err = s.FrontendClient().ListWorkflowExecutions(ctx, &workflowservice.ListWorkflowExecutionsRequest{
+			Namespace: s.Namespace().String(),
+			PageSize:  10,
+			Query:     query,
+		})
+		assert.NoError(t, err)
+		assert.NotNil(t, listResp)
+		if listResp != nil {
+			assert.Len(t, listResp.Executions, 2)
+		}
+	}, 5*time.Second, 500*time.Millisecond)
+
+	// unpause the activities in both workflows with batch unpause
+	_, err = s.SdkClient().WorkflowService().StartBatchOperation(context.Background(), &workflowservice.StartBatchOperationRequest{
+		Namespace: s.Namespace().String(),
+		Operation: &workflowservice.StartBatchOperationRequest_UnpauseActivitiesOperation{
+			UnpauseActivitiesOperation: &batchpb.BatchOperationUnpauseActivities{
+				Activity: &batchpb.BatchOperationUnpauseActivities_Type{Type: activityTypeName},
+			},
+		},
+		VisibilityQuery: fmt.Sprintf("WorkflowType='%s'", workflowTypeName),
+		JobId:           uuid.New(),
+		Reason:          "test",
+	})
+	s.NoError(err)
+
+	// make sure activities are unpaused
+	s.EventuallyWithT(func(t *assert.CollectT) {
+		description, err := s.SdkClient().DescribeWorkflowExecution(ctx, workflowRun1.GetID(), workflowRun1.GetRunID())
+		assert.NoError(t, err)
+		if description.GetPendingActivities() != nil {
+			assert.Len(t, description.PendingActivities, 1)
+			assert.False(t, description.PendingActivities[0].Paused)
+		}
+		description, err = s.SdkClient().DescribeWorkflowExecution(ctx, workflowRun2.GetID(), workflowRun2.GetRunID())
+		assert.NoError(t, err)
+		if description.GetPendingActivities() != nil {
+			assert.Len(t, description.PendingActivities, 1)
+			assert.False(t, description.PendingActivities[0].Paused)
+		}
+	}, 5*time.Second, 100*time.Millisecond)
+
+	// let both of the activities succeed
+	internalWorkflow.letActivitySucceed.Store(true)
+
+	var out string
+	err = workflowRun1.Get(ctx, &out)
+	s.NoError(err)
+
+	err = workflowRun2.Get(ctx, &out)
+	s.NoError(err)
+}
+
+func (s *ActivityApiBatchUnpauseClientTestSuite) TestActivityBatchUnpause_Negative() {
```

Comment:
I would be more concrete here too. Smth like:
```suggestion
func (s *ActivityApiBatchUnpauseClientTestSuite) TestActivityBatchUnpause_ValidationFailed() {
```

Reply:
renamed to "Failed"